doc/classes/Rect2i.xml

```diff
@@ -114,79 +136,93 @@
 			<return type="Rect2i" />
 			<param index="0" name="b" type="Rect2i" />
 			<description>
-				Returns the intersection of this [Rect2i] and [param b].
-				If the rectangles do not intersect, an empty [Rect2i] is returned.
+				Returns the intersection between this rectangle and [param b]. If the rectangles do not intersect, returns an empty [Rect2i].
+				[codeblocks]
+				[gdscript]
+				var a = Rect2i(0, 0, 5, 10)
+				var b = Rect2i(2, 0, 8, 4)
+
+				var c = a.intersection(b) # c is Rect2i(2, 0, 3, 4)
+				[/gdscript]
+				[csharp]
+				var a = new Rect2I(0, 0, 5, 10);
+				var b = new Rect2I(2, 0, 8, 4);
+
+				var c = rect1.Intersection(rect2); // c is Rect2I(2, 0, 3, 4)
+				[/csharp]
+				[/codeblocks]
+				[b]Note:[/b] If you only need to know whether two rectangles are overlapping, use [method intersects], instead.
 			</description>
 		</method>
 		<method name="intersects" qualifiers="const">
 			<return type="bool" />
 			<param index="0" name="b" type="Rect2i" />
 			<description>
-				Returns [code]true[/code] if the [Rect2i] overlaps with [param b] (i.e. they have at least one point in common).
+				Returns [code]true[/code] if this rectangle overlaps with the [param b] rectangle. The edges of both rectangles are excluded.
 			</description>
 		</method>
 		<method name="merge" qualifiers="const">
 			<return type="Rect2i" />
 			<param index="0" name="b" type="Rect2i" />
 			<description>
-				Returns a larger [Rect2i] that contains this [Rect2i] and [param b].
+				Returns a [Rect2i] that encloses both this rectangle and [param b] around the edges. See also [method encloses].
 			</description>
 		</method>
 	</methods>
 	<members>
 		<member name="end" type="Vector2i" setter="" getter="" default="Vector2i(0, 0)">
-			Ending corner. This is calculated as [code]position + size[/code]. Setting this value will change the size.
+			The ending point. This is usually the bottom-right corner of the rectangle, and is equivalent to [code]position + size[/code]. Setting this point also affects the [member size].
 		</member>
 		<member name="position" type="Vector2i" setter="" getter="" default="Vector2i(0, 0)">
-			Beginning corner. Typically has values lower than [member end].
+			The origin point. This is usually the top-left corner of the rectangle.
 		</member>
 		<member name="size" type="Vector2i" setter="" getter="" default="Vector2i(0, 0)">
-			Size from [member position] to [member end]. Typically, all components are positive.
-			If the size is negative, you can use [method abs] to fix it.
+			The rectangle's width and height, starting from [member position]. Setting this value also affects the [member end] point.
+			[b]Note:[/b] It's recommended setting the [member size] to a positive value, as most methods in Godot assume that the [member position] is the top-left corner, and the [member end] is the bottom-right corner. To get an equivalent rectangle with positive size, use [method abs].
```

Comment:
Suggestion: "It's recommended setting height and width of [member size] to non-negative values"